src/sys/system_v.rs

```diff
@@ -0,0 +1,385 @@
+/// assert_eq!(Permissions::new(512).expect_err("512 is bigger than what u9 can store"), Errno::E2BIG);
+/// # Ok(())
+/// # }
+/// ```
+pub struct Permissions {
+    permission: u16,
+}
+
+impl Permissions {
+    /// Create a new Permissions object
+    ///
+    /// Clamp to a u9 size, return Errno::E2BIG if it fails
+    ///
+    pub fn new(octal: u16) -> Result<Self> {
+        if octal >= 2_u16.pow(9) {
+            return Err(Errno::E2BIG);
+        }
+        Ok(Permissions { permission: octal })
+    }
+
+    /// Getter for permission
+    ///
+    pub fn get_permission(&self) -> &u16 {
+        &self.permission
+    }
+
+    /// Using the current stored permission, do a bitor operation on the
+    /// bitflags enums given
+    ///
+    pub fn to_octal<T: bitflags::Flags<Bits = i32>>(
+        &self,
+        vec_flags: Vec<T>,
+    ) -> c_int {
+        let mut flags: c_int = T::empty().bits();
+        for flag in vec_flags {
+            flags |= flag.bits();
+        }
+        flags |= self.permission as i32;
+        flags
+    }
+}
+
+libc_bitflags!(
+    /// Valid flags for the third parameter of the function [`shmget`]
+    pub struct ShmgetFlag: c_int
+    {
+        /// A new shared memory segment is created if key has this value.
+        IPC_PRIVATE;
+        /// Create a new segment.
+        /// If this flag is not used, then shmget() will find the segment
+        /// associated with key and check to see if the user has permission
+        /// to access the segment.
+        IPC_CREAT;
+        /// This flag is used with IPC_CREAT to ensure that this call creates
+        /// the segment.  If the segment already exists, the call fails.
+        IPC_EXCL;
+        /// Allocate the segment using "huge" pages.  See the Linux kernel
+        /// source file Documentation/admin-guide/mm/hugetlbpage.rst for
+        /// further information.
+        #[cfg(linux)]
+        SHM_HUGETLB;
+        // TODO: Does not exist in libc/linux, but should? Maybe open an issue in their repo
+        // SHM_HUGE_2MB;
+        // TODO: Same for this one
+        // SHM_HUGE_1GB;
+        /// This flag serves the same purpose as the mmap(2) MAP_NORESERVE flag.
+        /// Do not reserve swap space for this segment. When swap space is
+        /// reserved, one has the guarantee that it is possible to modify the
+        /// segment. When swap space is not reserved one might get SIGSEGV upon
+        /// a write if no physical memory is available. See also the discussion
+        /// of the file /proc/sys/vm/overcommit_memory in proc(5).
+        #[cfg(linux)]
+        SHM_NORESERVE;
+    }
+);
+/// Creates and returns a new, or returns an existing, System V shared memory
+/// segment identifier.
+///
+/// For more information, see [`shmget(2)`].
+///
+/// [`shmget(2)`]: https://man7.org/linux/man-pages/man2/shmget.2.html
+pub fn shmget(
+    key: key_t,
+    size: size_t,
+    shmflg: Vec<ShmgetFlag>,
```

Comment:
Do we need to use a `Vec` here, considering that `ShmgetFlag` is a bitflag so that it is ORable, for example, you can do:

```rs
shmget(key, size, ShmgetFlag::IPC_CREAT | ShmgetFlag::IPC_EXCL, permission)

Reply:
You're right, I forgot about that.
Fixed by changing to taking a `Flag` directly